File: DoItTest/ClientApp/src/domain/tests/items/answerOption/radioButtonAnswerOption.ts
import { TestItemType } from "../testItemType"
import { AnswerOption } from "./answerOption"

export class RadioButtonAnswerOption extends AnswerOption {
    constructor(
        public id: string,
        public testItemId: string,
        public type: TestItemType,
        public title: string,
        public isTrue: boolean
    ) {
        super(id, testItemId, type)
    }
}

export function mapRadioButtonAnswerOption(value: any): RadioButtonAnswerOption {
    return new RadioButtonAnswerOption(value.id, value.testItemId, value.type, value.title, value.isTrue)
}

export function mapRadioButtonAnswerOptions(value: any[]): RadioButtonAnswerOption[] {
    return value.map(mapRadioButtonAnswerOption)
}

File: DoItTest.Domain/Tests/TestItems/AnswerOptions/RadioButtonAnswerOption.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoItTest.Domain.Tests.TestItems.AnswerOptions
{
    public class RadioButtonAnswerOption : AnswerOption
    {
        public String Title { get; }
        public Boolean? IsTrue { get; }

        public RadioButtonAnswerOption(Guid id, Guid testItemId, TestItemType type, String title, Boolean? isTrue) : base(id, testItemId, type)
        {
            Title = title;
            IsTrue = isTrue;
        }
    }
}
